app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -371,26 +374,7 @@ class NewMessageFragment : Fragment() {
 
     private fun setupFromField(signatures: List<Signature>) = with(binding) {
 
-        val selectedSignatureId = newMessageViewModel.draftInRAM.identityId?.toInt() ?: -1
-        val selectedSignature = with(signatures) {
-            find { it.id == selectedSignatureId } ?: find { it.isDefault }!!
-        }
-        updateSelectedSignatureFromField(signatures.count(), selectedSignature)
-
-        val adapter = SignatureAdapter(
-            signatures = signatures,
-            selectedSignatureId = selectedSignatureId,
-            onClickListener = { newSelectedSignature ->
-                trackNewMessageEvent("switchIdentity")
-
-                updateSelectedSignatureFromField(signatures.count(), newSelectedSignature)
-                updateBodySignature(newSelectedSignature.content)
-
-                newMessageViewModel.draftInRAM.identityId = newSelectedSignature.id.toString()
-
-                addressListPopupWindow?.dismiss()
-            },
-        )
+        signatureAdapter.updateList(signatures)
```

Comment:
I don't like this "updateList" being only use as a setter
Everytime I'm asking myself what could be the value before